src/main/java/com/cloudbees/jenkins/support/SupportPlugin.java

```diff
@@ -668,6 +674,42 @@ public void onOnline(Computer c, TaskListener listener) throws IOException, Inte
                         "Could not install root log handler on node: " + c.getName(), e);
             }
         }
+
+        @Override
+        public void onOffline(@Nonnull final Computer c, @CheckForNull OfflineCause cause) {
+            if (Boolean.getBoolean(SupportPlugin.class.getName() + ".generateOnOfflineNode")) {
```

Comment:
Maybe makes sense to generate bundles only when the cause is not null and not RestartCause (which means the termination is abnormal). https://github.com/jenkinsci/jenkins/search?utf8=%E2%9C%93&q=%22extends+OfflineCause%22&type=Code . RestartCause just may produce much flood since it happens when Jenkins performs restart.

Kinda 🐜